scipy/signal/_peak_finding_utils.pyx

```diff
@@ -0,0 +1,72 @@
+"""
+Utility functions for finding peaks in signals.
+"""
+
+import numpy as np
+cimport numpy as np
+import cython
+
+
+__all__ = ['_argmaxima1d']
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def _argmaxima1d(np.ndarray[np.float64_t, ndim=1] x not None):
```

Comment:
Alternately, could specify `x`'s type as a memoryview: `np.float64_t[:]`. I seem to recall people saying that was preferred for new code, but I'm not sure.

Reply:
Good point. 
As I understand it I should introduce the following changes:
```python
def _argmaxima1d(np.float64_t[:] x not None):
    ...
    cdef np.int64_t[:] maxima = np.empty(x.shape[0] // 2, dtype=np.int64)
```
Cython's doc states that [memoryviews are considerably faster](http://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#comparison-to-the-old-buffer-support). However I don't see the expected results when comparing both approaches (see [here](http://nbviewer.jupyter.org/urls/gitlab.com/snippets/1696398/raw)).

I'd be interested to know what would be the most efficient way to initialize an empty array in Cython which later needs to be accessed by Python?

I'll try profiling this later.

Reply:
Turns out this seems to be quite hard to profile as the behavior changes significantly when profiling. I think I'll leave it as introduced here 23903fb in order to avoid premature optimization. ;)